pyro/ops/provenance.py

```diff
@@ -46,15 +46,21 @@ def __new__(cls, data: torch.Tensor, provenance=frozenset(), **kwargs):
         assert not isinstance(data, ProvenanceTensor)
         if not provenance:
             return data
-        return super().__new__(cls)
+        ret = data.view(data.shape)
+        ret._t = data.view(data.shape)  # this makes sure that detach_provenance always
```

Comment:
Could this line be simplified to
```python
ret._t = data
```
or would that break something?

Reply:
Right, thanks. Took me about four tries to get all the tests to pass, this was still a remnant of an earlier attempt.